src/services/ui/src/api/submissionService.ts

```diff
@@ -0,0 +1,147 @@
+import { API } from "aws-amplify";
+import {OnemacAttachmentSchema, Authority, ReactQueryApiError} from "shared-types";
+import {SubmissionServiceEndpoint} from "@/lib";
+import {OneMacUser} from "@/api/useGetUser";
+import {useMutation, UseMutationOptions} from "@tanstack/react-query";
+
+export enum SubmissionVariant {
+  INITIAL = "initial",
+  ACTION = "action"
+}
+type SubmissionServiceParameters<T> = {
+  data: T,
+  endpoint: SubmissionServiceEndpoint
+  variant: SubmissionVariant,
+  user: OneMacUser | undefined,
+  authority?: Authority
+}
+type SubmissionServiceResponse = {
+  body: {
+    message: string;
+  };
+};
+type PreSignedURL = {
+  url: string;
+  key: string;
+  bucket: string;
+};
+type UploadRecipe = PreSignedURL & {
+  data: File[],
+  title: string
+}
+type AttachmentsSchema = Record<string, File[] | undefined>
+
+/** Grabs one pre-signed URL for file upload */
+const getPreSignedUrl = async (): Promise<PreSignedURL> => {
+  // TODO: Can this be a GET instead of POST w/ empty body
+  return await API.post("os", "/getUploadUrl", {
+    body: {},
```

Comment:
@mdial89f Can this be a GET? POST with an empty body seems like we're maybe compensating for an external services requirement, and if that isn't the case, I think we should refactor.